huaweicloud/resource_huaweicloud_compute_instance.go

```diff
@@ -681,27 +681,45 @@ func resourceComputeInstanceV2Create(ctx context.Context, d *schema.ResourceData
 		}
 	}
 
-	if err := resourceComputeInstanceV2Read(ctx, d, meta); err != nil {
-		return err
-	}
+	// get the original value of source_dest_check in script
+	originalNetworks := d.Get("network").([]interface{})
+	sourceDestChecks := make([]bool, len(originalNetworks))
+	var flag bool
 
-	// note: as "network.port" is optional, we shoud use it after resourceComputeInstanceV2Read
-	networks := d.Get("network").([]interface{})
-	for _, v := range networks {
+	for i, v := range originalNetworks {
 		nic := v.(map[string]interface{})
-		nicPort := nic["port"].(string)
-		if nicPort == "" {
-			continue
+		sourceDestChecks[i] = nic["source_dest_check"].(bool)
+		if !sourceDestChecks[i] {
```

Comment:
After `flag` has been setted for the first time, it doesn‘t need to be set subsequently.
`if !flag && !sourceDestCheck[i] {`

Reply:
fixed